libcudacxx/include/cuda/std/__type_traits/num_bits.h

```diff
@@ -0,0 +1,96 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCUDACXX___TYPE_TRAITS_NUM_BITS
+#define _LIBCUDACXX___TYPE_TRAITS_NUM_BITS
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__complex/is_complex.h>
+#include <cuda/std/__type_traits/always_false.h>
+#include <cuda/std/__type_traits/enable_if.h>
+#include <cuda/std/__type_traits/has_unique_object_representation.h>
+#include <cuda/std/__type_traits/is_extended_floating_point.h>
+#include <cuda/std/__type_traits/is_floating_point.h>
+#include <cuda/std/__type_traits/is_same.h>
+#include <cuda/std/__type_traits/remove_cv.h>
+#include <cuda/std/climits>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+template <typename _Tp>
+struct _PreventInstantiation
+{
+  static_assert(__always_false_v<_Tp>);
+};
+
+template <typename _Tp, typename = remove_cv_t<_Tp>>
+inline constexpr int __num_bits_v = _PreventInstantiation<_Tp>{};
+
+template <typename _Tp>
+inline constexpr int __num_bits_v<_Tp,
```

Comment:
This is overly complex and I would rather have this either as separate specializations of `__num_bits_v` (preferably, given that we specialize it below) or as a constexpr function with `if constexpr` switches

Why do we include `complex` and not `pair` or arrays. If the goal is to expand this to aggregates, then we can consider using this for anything with static `tuple_size`, however, I am not sure of the use case here